src/docfx/lib/log/CustomRule.cs

```diff
@@ -30,5 +37,22 @@ public CustomRule(ErrorLevel? severity, string? code, string? additionalMessage,
             CanonicalVersionOnly = canonicalVersionOnly;
             PullRequestOnly = pullRequestOnly;
         }
+
+        public Func<string, bool> GetCachedGlobMatcher(string glob)
```

Comment:
I added 100+ exclusion paths and noticed marginal impact, so I added the cache. Every little bit helps!